components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/inbound/websocket/utils/InboundWebsocketProcessorUtil.java

```diff
@@ -0,0 +1,711 @@
+                                                        InboundMessageContext inboundMessageContext) {
+
+        if (authenticationContext == null || !authenticationContext.isAuthenticated()) {
+            return false;
+        }
+        // The information given by the AuthenticationContext is set to an APIKeyValidationInfoDTO object
+        // so to feed information analytics and throttle data publishing
+        APIKeyValidationInfoDTO info = new APIKeyValidationInfoDTO();
+        info.setAuthorized(authenticationContext.isAuthenticated());
+        info.setApplicationTier(authenticationContext.getApplicationTier());
+        info.setTier(authenticationContext.getTier());
+        info.setSubscriberTenantDomain(authenticationContext.getSubscriberTenantDomain());
+        info.setSubscriber(authenticationContext.getSubscriber());
+        info.setStopOnQuotaReach(authenticationContext.isStopOnQuotaReach());
+        info.setApiName(authenticationContext.getApiName());
+        info.setApplicationId(authenticationContext.getApplicationId());
+        info.setType(authenticationContext.getKeyType());
+        info.setApiPublisher(authenticationContext.getApiPublisher());
+        info.setApplicationName(authenticationContext.getApplicationName());
+        info.setConsumerKey(authenticationContext.getConsumerKey());
+        info.setEndUserName(authenticationContext.getUsername());
+        info.setApiTier(authenticationContext.getApiTier());
+        info.setGraphQLMaxDepth(authenticationContext.getGraphQLMaxDepth());
+        info.setGraphQLMaxComplexity(authenticationContext.getGraphQLMaxComplexity());
+
+        inboundMessageContext.setKeyType(info.getType());
+        inboundMessageContext.setInfoDTO(info);
+        inboundMessageContext.setAuthContext(authenticationContext);
+        inboundMessageContext.setInfoDTO(info);
+        return authenticationContext.isAuthenticated();
+    }
+
+    /**
+     * Authenticates JWT token in incoming GraphQL subscription requests.
+     *
+     * @param inboundMessageContext InboundMessageContext
+     * @return true if authenticated
+     * @throws APIManagementException if an internal error occurs
+     * @throws APISecurityException   if authentication fails
+     */
+    public static boolean authenticateGraphQLJWTToken(InboundMessageContext inboundMessageContext)
+            throws APIManagementException, APISecurityException {
+
+        AuthenticationContext authenticationContext;
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(inboundMessageContext.getTenantDomain(),
+                true);
+        JWTValidator jwtValidator = new JWTValidator(new APIKeyValidator(), inboundMessageContext.getTenantDomain());
+        authenticationContext = jwtValidator.
+                authenticateForGraphQLSubscription(inboundMessageContext.getSignedJWTInfo(),
```

Comment:
Here as well, Can't we pass only inboundMessageContext?

Reply:
IMO JWTValidator logic should not depend on the transport specific message context InboundMessageContext or SynapseMessageContext, hence passing the signed JWT directly is better.

Reply:
+1